### PR TITLE
feat: [CDS-76035]: Replace time_bucket functionality for timescale migration for CDOverviewDashboardServiceImpl

### PR DESCRIPTION
**Pull request by mani5h-harness**
## Describe your changes
This PR adds harness_date_bin_ng_mgr function which is the replacement for time_bucket functionality in CDOverviewDashboardServiceImpl

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52529)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java [hunk 1]
```diff
@@ -1956,7 +1956,7 @@ public String queryBuilderServiceDeployments(String accountIdentifier, String or
         "select status, time_entity, COUNT(*) as numberOfRecords from (select service_status as status, service_startts as execution_time, ";
     totalBuildSqlBuilder.append(selectQuery)
         .append(String.format(
-            "time_bucket(%s, service_startts) as time_entity, pipeline_execution_summary_cd_id  from service_infra_info as sii, pipeline_execution_summary_cd as pesi where pesi.accountid = sii.accountid AND sii.service_id is not null and ",
+            "harness_date_bin_ng_mgr(%s, service_startts) as time_entity, pipeline_execution_summary_cd_id  from service_infra_info as sii, pipeline_execution_summary_cd as pesi where pesi.accountid = sii.accountid AND sii.service_id is not null and ",
```
Comment (RishabhGupta34) [inline 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java:1959]: @mani5h-harness can you confirm the places this query is being used?

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52529)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by harness-david — APPROVED** [review]
Changes are straightforward.  It's a relatively easy drop-in replacement.  Thanks.

**Review by RishabhGupta34 — APPROVED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java (hunk 1 above)
Comment (mani5h-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/overview/service/CDOverviewDashboardServiceImpl.java:1959]: as we checked offline, 
there are three paths where this gets hit

serviceDeploymentsInfo
serviceDeploymentsInfoV2
serviceDeployments

only serviceDeploymentsInfoV2 is being used now, which I have verified

**Comment by mani5h-harness** [comment]
trigger uts

**Comment by mani5h-harness** [comment]
trigger review

**Pull request merged**